### ISSUE TITLE
Two RandomFlip layers with same seed do not return the same result

### ISSUE THREAD

**Issue by cofri** [keras-team/tf-keras#287]
**System information**.
- Have I written custom code: No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04
- TensorFlow installed from (source or binary): pip install
- TensorFlow version (use command below): 2.9.2 / 2.10.1 / 2.11.0
- Python version: 3.8

**Describe the problem**.

With TensorFlow 2.10+, two `RandomFlip` layers instantiated with the same seed do not return the same result. For instance, when applying the two random flip operations on an image, I expect that the two augmented images are both flipped or not, with the same output. However, this is not the case: on multiple runs, the two augmented images are not always equal (see standalone code below).

Note that with TensorFlow 2.9.2, the expected behaviour is observed: same result for both RandomFlip layers. Since TensorFlow 2.10, this is not the case anymore.

Note also that for two `RandomRotation` layers with the same seed, the same rotation is applied on the image, whatever the version of TensorFlow (2.9, 2.10 and 2.11). This seems to be the expected behaviour, compared to the RandomFlip behaviour.

**Standalone code to reproduce the issue**.
```
import tensorflow as tf
print(tf.__version__)

x = tf.convert_to_tensor([[[1], [0]]]) # Tensor of shape (1, 2, 1)

# Create two RandomFlip layers with same seed
seed = 42
flip1 = tf.keras.layers.RandomFlip(mode="horizontal", seed=seed)
flip2 = tf.keras.layers.RandomFlip(mode="horizontal", seed=seed)

# Run 100 times random flip with the two layers
# In TensorFlow 2.9, same result every time (100 %)
# In TensorFlow 2.10+, not the same result every time
total = 0
N = 100
for i in range(N):
    y1 = flip1(x)
    y2 = flip2(x)
    if tf.math.reduce_all(tf.equal(y1, y2)):
        total += 1
print(f"y1 / y2 equal? {total / N * 100} %")
```

**Comment by michaeldiener** [comment]
I'm using TF 2.14 have the same problem, but for RandomRotation instead. I've been digging through the code and found the issue and a workaround.
The problem seems to be the random generator being stateful instead of stateless. See the [source](https://github.com/keras-team/tf-keras/blob/ff2ef2100695af2d67dc37a8992ea4b942088adb/tf_keras/backend.py#L1888) for an explanation.

So my workaround is:
`seedRotation = 55
rotate1 = layers.RandomRotation(1, seed=seedRotation)
rotate1._random_generator._set_rng_type(rng_type="stateless")
rotate2 = layers.RandomRotation(1, seed=seedRotation)
rotate2._random_generator._set_rng_type(rng_type="stateless")
`

Should work the same for RandomFlip as they both have the same super class (or whatever you call that in python) that initializes the random generator. 

PS: Yes, the workaround is a bit hacky, but it works and solves the issue for now.
